SEED-platform/seed
# Meters merge update

#### Any background context you want to provide?
Meters and subsequent MeterReadings could potentially be lost when merging and unmerging Properties manually. 

#1894 

#### What's this PR do?
The following has been implemented:
- Users are warned when any properties being merged have associated Meters.
- Merge Meters and Readings if any or all properties being merged have them. In the case of merging overlapping Readings (by start_time and end_time), priority is given to Readings belonging to the 'priority' Property.
- During unmerging, Meters are copied to both of the new (canonical) Properties.

#### How should this be manually tested?
Via Inventory List and Detail Property pages:
- Add overlapping and non-overlapping meters to properties.
- Check that ONLY when merging properties with meters, a warning pops up. This should not happen if none of the selected have Meters or taxlots are being merged.
- Merge while checking that meter readings persist and/or are combined.
- Unmerge while checking that both of the new properties now have identical readings.
- When merging overlapping readings, be sure that the priority order between properties is inherited amongst the meter readings.  Properties selected most recently get more priority, so their readings should be used when overlaps occur.
